framework/include/actions/Action.h

```diff
@@ -44,8 +46,48 @@ class Action : public ConsoleStreamInterface
 
   virtual void act() = 0;
 
+  /**
+   * The name of the action
+   *
+   * When using MOOSE input files, this is the last portion
+   * of the full input syntax:
```

Comment:
Well - yes and no.  The name is just the danged name of the object... regardless of whether you're using MOOSE input file syntax or not.